x/dex/handler.go

```diff
@@ -40,3 +39,18 @@ func NewHandler(k keeper.Keeper, tracingInfo *tracing.TracingInfo) sdk.Handler {
 		}
 	}
 }
+
+func NewRegisterPairsProposalHandler(k keeper.Keeper) govtypes.Handler {
+	return func(ctx sdk.Context, content govtypes.Content) error {
+		switch c := content.(type) {
+		case *types.RegisterPairsProposal:
+			return handleRegisterPairsProposal(ctx, k, c)
+		default:
+			return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unrecognized dex proposal content type: %T", c)
+		}
+	}
+}
+
+func handleRegisterPairsProposal(ctx sdk.Context, k keeper.Keeper, p *types.RegisterPairsProposal) error {
```

Comment:
why do we need this function? why don't we just call
`k.HandleRegisterPairsProposal(ctx, p)` in the switch-case?

Reply:
Updated